crates/holochain_state/src/source_chain.rs

```diff
@@ -856,11 +860,11 @@ where
         Ok(query.filter_records(records))
     }
 
-    pub async fn is_chain_locked(&self, lock: Vec<u8>) -> SourceChainResult<bool> {
+    pub async fn is_chain_locked(&self) -> SourceChainResult<Option<Vec<u8>>> {
```

Comment:
This is a surprising signature. I'd rename the function or return whether the subject is some or none => bool
```suggestion
    pub async fn get_chain_lock_subject(&self) -> SourceChainResult<Option<Vec<u8>>> {
```